src/tabs.ts

```diff
@@ -0,0 +1,412 @@
+      if (bodyMain) return bodyMain
+      const newBodyMain = ele.bodyTR!.appendChild(dom.createElement('main'))
+      newBodyMain.setAttribute('style', bodyMainStyle)
+      return newBodyMain
+    }
+  }
+
+  // @@ Use common one from utils?
+  function orderedSync () {
+    const items = getItems()
+    let slot: TabElement, i, j, left, right
+    let differ = false
+    // Find how many match at each end
+    for (left = 0; left < tabContainer.children.length; left++) {
+      slot = tabContainer.children[left] as TabElement
+      if (
+        left >= items.length ||
+        (slot.subject && !slot.subject.sameTerm(items[left]))
+      ) {
+        differ = true
+        break
+      }
+    }
+    if (!differ && items.length === tabContainer.children.length) {
+      return // The two just match in order: a case to optimize for
+    }
+    for (right = tabContainer.children.length - 1; right >= 0; right--) {
+      slot = tabContainer.children[right] as TabElement
+      j = right - tabContainer.children.length + items.length
+      if (slot.subject && !slot.subject.sameTerm(items[j])) {
+        break
+      }
+    }
+    // The elements left ... right in tabContainer.children do not match
+    const insertables = items.slice(left, right - tabContainer.children.length + items.length + 1)
+    while (right >= left) {
+      // remove extra
+      tabContainer.removeChild(tabContainer.children[left])
+      bodyContainer.removeChild(bodyContainer.children[left])
+      right -= 1
+    }
+    for (i = 0; i < insertables.length; i++) {
+      const newSlot = makeNewSlot(insertables[i])
+      const newBodyDiv = dom.createElement('div')
+      newSlot.bodyTR = newBodyDiv
+      if (left === tabContainer.children.length) {
+        // None left of original on right
+        tabContainer.appendChild(newSlot)
+        bodyContainer.appendChild(newBodyDiv)
+      } else {
+        tabContainer.insertBefore(newSlot, tabContainer.children[left + i])
+        bodyContainer.insertBefore(newBodyDiv, bodyContainer.children[left + i])
+      }
+    }
+    if (onClose) {
+      addCancelButton(tabContainer)
+    }
+  }
+
+  function resetTabStyle () {
+    for (let i = 0; i < tabContainer.children.length; i++) {
+      const tab = tabContainer.children[i]
+      if (tab.classList.contains('unstyled')) {
+        continue
+      }
+      if (tab.children[0]) {
+        tab.children[0].setAttribute('style', unselectedStyle)
+      }
+    }
+  }
+
+  function resetBodyStyle () {
+    for (let i = 0; i < bodyContainer.children.length; i++) {
+      bodyContainer.children[i].setAttribute('style', hiddenStyle)
+    }
+  }
+}
+
+/**
+ * @internal
+ */
+function getColors (backgroundColor: string): [string, string] {
+  return isLight(backgroundColor)
+    ? [colorBlend(backgroundColor, '#ffffff', 0.3), '#000000']
+    : [colorBlend(backgroundColor, '#000000', 0.3), '#ffffff']
+}
+
+/**
+ * @internal
```

Comment:
just wondering why you used @internal instead of @ignore? I can't seem to find anything on @internal to see why to use one versus the other.

Reply:
There's a [section on @internal in the TypeDoc guides](https://typedoc.org/guides/doccomments/#internal), and I thought the distinction might be good to have (they are, after all, internal functions that should not be exposed on the public API), especially since TypeDoc has the option `stripInternal`.

Now, if we want to describe internal APIs as well at some point, we can just remove the option `stripInternal` when generating the docs.